pkg/controller/provider/controller.go

```diff
@@ -334,6 +369,140 @@ func (r *Reconciler) getSecret(provider *api.Provider) (*core.Secret, error) {
 	return secret, nil
 }
 
+func (r *Reconciler) createOVAServerDeployment(provider *api.Provider, ctx context.Context) {
+
+	deploymentName := fmt.Sprintf("%s-deployment-%s", ovaServerPrefix, provider.Name)
+	annotations := make(map[string]string)
+	labels := map[string]string{"providerName": provider.Name, "app": "forklift"}
+	url := provider.Spec.URL
+	var replicas int32 = 1
+
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "forklift.konveyor.io/v1beta1",
+		Kind:       "Provider",
+		Name:       provider.Name,
+		UID:        provider.UID,
+	}
+
+	//OVA server deployment
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            deploymentName,
+			Namespace:       provider.Namespace,
+			Annotations:     annotations,
+			Labels:          labels,
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "forklift",
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"providerName": provider.Name,
+						"app":          "forklift",
+					},
+				},
+				Spec: makeOvaProviderPodSpec(url, string(provider.UID)),
+			},
+		},
+	}
+
+	err := r.Create(ctx, deployment)
+	if err != nil {
+		return
+	}
+
+	// OVA Server Service
+	serviceName := fmt.Sprintf("ova-service-%s", provider.Name)
+	service := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            serviceName,
+			Namespace:       provider.Namespace,
+			Labels:          labels,
+			OwnerReferences: []metav1.OwnerReference{ownerReference},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{
+				"providerName": provider.Name,
+				"app":          "forklift",
+			},
+			Ports: []v1.ServicePort{
+				{
+					Name:       "api-http",
+					Protocol:   v1.ProtocolTCP,
+					Port:       8080,
+					TargetPort: intstr.FromInt(8080),
+				},
+			},
+			Type: v1.ServiceTypeClusterIP,
+		},
+	}
+
+	err = r.Create(ctx, service)
+	if err != nil {
+		return
+	}
+}
+
+func makeOvaProviderPodSpec(url string, id string) corev1.PodSpec {
```

Comment:
```suggestion
func makeOvaProviderPodSpec(url string, providerId string) corev1.PodSpec {
```
shall we use the provider name here as well?